tests/lib/newlib/newlib_hook/src/main.c

```diff
@@ -0,0 +1,177 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <sys/unistd.h>
+#include <sys/signal.h>
+#include <sys/stat.h>
+
+/**
+ * @brief Verify _open() stub in newlib libc-hooks.c
+ *
+ * @details The compiler warning when call open(),
+ * use fopen() instead.
+ */
+void test_newlib_stub_open(void)
+{
+	FILE *fp = NULL;
+
+	fp = fopen("../prj.conf", "r");
+	zassert_is_null(fp, "open file error");
+}
+
+/**
+ * @brief Verify _close() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return -1.
+ */
+void test_newlib_stub_close(void)
+{
+	int ret = 0;
+
+	ret = close(-1);
+	zassert_equal(ret, -1, "closed a invalid fd");
+}
+
+/**
+ * @brief Verify _getpid() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return 0.
+ */
+void test_newlib_stub_getpid(void)
+{
+	pid_t pid;
+
+	pid = getpid();
+	zassert_false(pid, "getpid failed");
+}
+
+/**
+ * @brief Verify _stat() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return 0.
+ */
+void test_newlib_stub_fstat(void)
+{
+	struct stat status;
+	int ret;
+
+	ret = fstat(-1, &status);
+	zassert_false(ret, "fstat failed");
+}
+
+/**
+ * @brief Verify _kill() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return 0.
+ */
+void test_newlib_stub_kill(void)
+{
+	int ret = 0;
+
+	ret = raise(SIGILL);
+	zassert_false(ret, "kill failed");
+}
+
+/**
+ * @brief Verify _read() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return nbytes get.
+ */
+void test_newlib_stub_read(void)
+{
+	int ret;
+	char buf[100];
+
+	ret = read(-1, buf, 10);
+	zassert_true(ret, "read failed");
+}
+
+/**
+ * @brief Verify _write() stub in newlib libc-hooks.c
+ *
+ * @details Use character output functions to test _write().
+ */
+void test_newlib_stub_write(void)
+{
+	int ret;
+	int i = 3;
+	char buf[100] = {0};
+
+	ret = write(-1, buf, 10);
+	zassert_equal(ret, 10, "printf failed!");
+	ret = printf("%d\n", 3);
+	zassert_equal(ret, 2, "printf failed!");
+	ret = fprintf(stdout, "%d\n", i);
+	zassert_equal(ret, 2, "fprintf failed!");
+	ret = fputs("This 3\n", stdout);
+	zassert_equal(ret, 0, "fputs \"This 3\" failed");
+	ret = puts("This 3");
+	zassert_true(ret > 0, "puts \"This 3\" failed");
+	ret = fputc('T', stdout);
+	zassert_equal(ret, 84, "fputc \'T\' failed");
+	ret = putc('T', stdout);
+	zassert_equal(ret, 84, "putc \'T\' failed");
+}
+
+/**
+ * @brief Verify _sbrk() stub in newlib libc-hooks.c
+ *
+ * @details Use malloc and free to test _sbrk().
+ */
+void test_newlib_stub_sbrk(void)
+{
+	char *ptr = malloc(10);
+
+	zassert_not_null(ptr, "malloc failed");
+	free(ptr);
+}
+
+/**
+ * @brief Verify _lseek() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return 0.
+ */
+void test_newlib_stub_lseek(void)
+{
+	int ret;
+
+	ret = lseek(-1, 0, SEEK_END);
+	zassert_false(ret, "lseek failed");
```

Comment:
`ret` is not a boolean type.
```suggestion
	zassert_equal(ret, 0, "lseek failed");
```